test/TEST-08-ISSUE-2730/test.sh

```diff
@@ -23,8 +23,11 @@ check_result_qemu() {
 }
 
 test_run() {
-    run_qemu || return 1
-    check_result_qemu || return 1
+    if run_qemu; then
+        check_result_qemu || return 1
+    else
```

Comment:
I did test that it works with and without QEMU. Can you please be more specific about "doesn't work", so that I can reproduce the problem with that? Thanks!